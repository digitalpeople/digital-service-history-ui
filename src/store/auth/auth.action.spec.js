import {
  loginRequest,
  loginSuccess,
  loginFailure,
  logout,
} from './auth.action';

describe('auth.action', () => {
  describe('loginRequest', () => {
    it('should return action', () => {
      const actual = loginRequest();

      expect(actual).toMatchSnapshot();
    });
  });
  describe('loginSuccess', () => {
    it('should return action', () => {
      const userId = 'test-user-id';

      const actual = loginSuccess(userId);

      expect(actual).toMatchSnapshot();
    });
  });
  describe('loginFailure', () => {
    it('should return action', () => {
      const error = {
        message: 'test-error-message',
      };

      const actual = loginFailure(error);

      expect(actual).toMatchSnapshot();
    });
  });
  describe('logout', () => {
    it('should return action', () => {
      const actual = logout();

      expect(actual).toMatchSnapshot();
    });
  });
});
