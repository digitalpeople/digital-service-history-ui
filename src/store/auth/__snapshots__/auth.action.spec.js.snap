// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth.action loginFailure should return action 1`] = `
Object {
  "error": Object {
    "message": "test-error-message",
  },
  "type": "@auth/LOGIN_FAILURE",
}
`;

exports[`auth.action loginRequest should return action 1`] = `
Object {
  "type": "@auth/LOGIN_REQUEST",
}
`;

exports[`auth.action loginSuccess should return action 1`] = `
Object {
  "type": "@auth/LOGIN_SUCCESS",
  "userId": "test-user-id",
}
`;

exports[`auth.action logout should return action 1`] = `
Object {
  "type": "@auth/LOGOUT",
}
`;
